src/torch/use_cases.py

```diff
@@ -0,0 +1,269 @@
+        if len(PrimalConvBlocks) != len(DualConvBlocks):
+            raise ValueError("Length of Primal and Dual Conv Blocks must be the same")
+        self.PrimalConvBlocks = torch.nn.ModuleList(PrimalConvBlocks)
+        self.DualConvBlocks = torch.nn.ModuleList(DualConvBlocks)
+        # add sirf singleton for channel then remove sirf singletons acq data
+        self.FwdOperator = lambda x: FwdOperator(x).squeeze(2).squeeze(3)
+        # add sirf singleton for channel then remove sirf singleton 2D image
+        self.AdjOperator = lambda x: AdjOperator(x.unsqueeze(2).unsqueeze(2)).squeeze(2)
+        self.relu = torch.nn.ReLU()
+    
+    def forward(self, y):
+
+        batch_size = y.shape[0]
+        h = torch.zeros_like(y, device=y.device)
+        dummy_f = self.AdjOperator(y)
+        f = torch.zeros_like(dummy_f, device=y.device)
+        n_iter = len(self.PrimalConvBlocks)
+
+        for i in range(n_iter):
+            Op_f = self.FwdOperator(f)
+            dual_input = torch.cat([Op_f, y], dim=1)
+            h = self.DualConvBlocks[i](dual_input)
+            h = self.relu(h)
+            OpAdj_h = self.AdjOperator(h)
+            f = self.PrimalConvBlocks[i](torch.cat([f,OpAdj_h], dim=1)) 
+            f = self.relu(f) 
+        
+        return f
+
+
+
+
+class UseCases:
+    def __init__(self, modality, dim, test_name):
+        self.device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+        if modality == 'PET':
+            if dim == '2D':
+                self.acq_data, self.image, self.acq_model = get_pet_2d()
+                # make torch versions of acq_data and image
+                self.torch_acq_data = sirf_to_torch(self.acq_data, self.device)
+                self.torch_image = sirf_to_torch(self.image, self.device).unsqueeze(0)
+            elif dim == '3D':
+                self.acq_data, self.image, self.acq_model = get_pet_3d()
+        if test_name == 'all':
+            if dim == '2D':
+                self.pet_varnet()
+                self.learned_primal_dual()
+            self.compare_gradient_descents()
+        elif test_name == 'gradient_descent':
+            self.compare_gradient_descents()
+        elif test_name == 'unrolled_backward':
+            self.unrolled_backward_test()
+        elif test_name == 'objective_function':
+            self.objective_function_test()
+
+
+
+    def learned_primal_dual(self):
+        # use seed for reproducibility
+
+        # set up learned primal dual network
+        PrimalConvBlocks = [ConvBlock() for i in range(3)]
+        DualConvBlocks = [ConvBlock() for i in range(3)]
+        # set up the forward and adjoint operators
+        sirf_image_template = self.image.get_uniform_copy(1)
+        FwdOperator = SIRFTorchOperator(self.acq_model, self.image.get_uniform_copy(1))
+        AdjOperator = SIRFTorchOperator(sirf.SIRF.AdjointOperator(self.acq_model.get_linear_acquisition_model()), self.acq_data.get_uniform_copy(1))
```

Comment:
I'm suprised by the `sirf.SIRF.AdjointOperator`. can't we use `sirf.AdjointOperator`? If not, shouldn't we be able to? (presumably some stuff to do with __init__.py)